packages/cohort/src/replicator/replicator_service.rs

```diff
@@ -0,0 +1,127 @@
+use std::{fmt::Debug, time::Duration};
+
+use crate::replicator::core::{DecisionOutcome, StateMapItem};
+
+use super::{
+    core::{CandidateMessage, ReceiverMessage},
+    suffix::SuffixDecisionTrait,
+};
+use log::{info, warn};
+use talos_certifier::{model::DecisionMessageTrait, ports::MessageReciever};
+use talos_suffix::{SuffixItem, SuffixTrait};
+
+pub struct Replicator<S, M>
+where
+    S: SuffixTrait<CandidateMessage> + SuffixDecisionTrait<CandidateMessage> + Debug,
+    M: MessageReciever<Message = ReceiverMessage> + Send + Sync,
+{
+    receiver: M,
+    suffix: S,
+}
+
+impl<S, M> Replicator<S, M>
+where
+    S: talos_suffix::SuffixTrait<CandidateMessage> + SuffixDecisionTrait<CandidateMessage> + Debug,
+    M: MessageReciever<Message = ReceiverMessage> + Send + Sync,
+{
+    pub fn new(receiver: M, suffix: S) -> Self {
+        Replicator { receiver, suffix }
+    }
+
+    pub async fn run(&mut self) {
```

Comment:
Think about splitting this into smaller methods for easier maintenance and unit testing later on.